packages/mcp/src/client.ts

```diff
@@ -1,14 +1,21 @@
 import { createTool } from '@mastra/core/tools';
 import { jsonSchemaToModel } from '@mastra/core/utils';
 import { Client } from '@modelcontextprotocol/sdk/client/index.js';
+import { SSEClientTransport } from '@modelcontextprotocol/sdk/client/sse.js';
 import { StdioClientTransport } from '@modelcontextprotocol/sdk/client/stdio.js';
 import type { StdioServerParameters } from '@modelcontextprotocol/sdk/client/stdio.js';
-import { ListResourcesResultSchema } from '@modelcontextprotocol/sdk/types.js';
+import type { Protocol } from '@modelcontextprotocol/sdk/shared/protocol.js';
+import type { Transport } from '@modelcontextprotocol/sdk/shared/transport.js';
 import type { ClientCapabilities } from '@modelcontextprotocol/sdk/types.js';
+import { ListResourcesResultSchema } from '@modelcontextprotocol/sdk/types.js';
+
+type SSEClientParameters = {
+  url: URL;
+} & ConstructorParameters<typeof SSEClientTransport>[1];
 
 export class MastraMCPClient {
```

Comment:
Don't you need to extend from MastraBase and create a abastract class so people can extend

Reply:
This client existed but only supported Stdio MCP servers before this PR - there are two main supported transports: sse and stdio

Reply:
There's also websockets and custom transports but those are less official